test/cvs-scm/run.sh

```diff
@@ -0,0 +1,94 @@
+#!/bin/bash -e
+#
+#  Quick functionality test for CVS
+#
+. ../test-lib.sh 2>/dev/null || { echo "Must run in script directory!" ; exit 1 ; }
+
+dir=$(mktemp -d)
+echo "Using scratch dir: $dir"
+
+# Bob recipes
+bob=$dir/bob
+mkdir -p "$bob/recipes"
+cat >"$bob/recipes/t.yaml" <<EOF
+root: true
+checkoutSCM:
+  scm: cvs
+  cvsroot: \${CVSROOT}
+  module: mod
+buildScript: |
+  cp "\$1/file.txt" .
+packageScript: |
+  cp "\$1/file.txt" .
+EOF
+
+# Directory to play in
+work=$dir/work
+mkdir "$work"
+echo init > "$work/file.txt"
+
+# CVS repository: contains a single repository "mod" with our file
+cvsroot=$dir/cvsroot
+mkdir "$cvsroot"
+cvs -d "$cvsroot" init
+(
+  cd "$work"
+  cvs -d "$cvsroot" import -m "message" mod vendor release
+  cvs -d "$cvsroot" co -d . mod
+)
+  
+
+##
+##  First check: check out initial state
+##
+run_bob -C "$bob" dev -DCVSROOT="$cvsroot" t
+result=$(run_bob -C "$bob" query-path -DCVSROOT="$cvsroot" -f {dist} t)
+test -n "$result"
+diff -q "$bob/$result/file.txt" "$work/file.txt"
+
+
+##
+##  Second check: modify the file and check that it is picked up by bob
+##  (incremental update works)
+##
+echo modif > "$work/file.txt"
+(
+  cd "$work"
+  cvs ci -m "update message" file.txt
+)
+
+run_bob -C "$bob"  dev -DCVSROOT="$cvsroot" t
+diff -q "$bob/$result/file.txt" "$work/file.txt"
+
+
+##
+##  Third check: commit to a branch and checkout that
+##  (change of branch/tag works)
+##
+echo branch > "$work/file.txt"
+(
+  cd "$work"
+  cvs tag -b my_branch
+  cvs up  -r my_branch
+  cvs ci -m "branch message" file.txt
+)
+
+cat >"$bob/recipes/t.yaml" <<EOF
+root: true
+checkoutSCM:
+  scm: cvs
+  cvsroot: \${CVSROOT}
+  module: mod
+  rev: my_branch
+buildScript: |
+  cp "\$1/file.txt" .
+packageScript: |
+  cp "\$1/file.txt" .
+EOF
+
+run_bob -C "$bob" dev -DCVSROOT="$cvsroot" t
+diff -q "$bob/$result/file.txt" "$work/file.txt"
+
+# Clean up
+rm -rf "$dir"
+cleanup
```

Comment:
The `rm` is not needed if the `trap` is used above. The `cleanup` should actually be done as first step in the test to evict any stale state from the last run.